### PR TITLE
Improve KFServing main README

### LINKED ISSUE #602: installation instructions need to accommodate the KF0.7 installation of knative-serving
/kind bug

It looks like the kf0.7 install (at least via the c2d app) includes installation of some version of knative-serving.  So the instructions here: https://github.com/kubeflow/kfserving/blob/master/docs/DEVELOPER_GUIDE.md#install-knative-on-a-kubernetes-cluster  ("For Kubeflow install, you can install Knative Serving v0.8 via the following commands after downloading repository kubeflow/manifests.") would seem to be outdated, as those manifests can not be successfully deployed.

### PR DESCRIPTION
**Pull request by yuzisun**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #602 

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note

```

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kfserving/656)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -31,7 +39,72 @@ env:
...
+sklearn-iris   http://sklearn-iris.default.example.com/v1/models/sklearn-iris   True    100                                109s
+```
+### KFServing Concepts and Data Plane
+[KFServing Concepts and Data Plane](./docs/README.md)
+
+
+### KFServing API Reference
+[KFServing API Docs](./docs/apis/README.md)
+
+### Examples
```
Comment (animeshsingh) [inline README.md]: This would be an overkill here. General guidance for Kubeflow repos is to shift more detailed documentation to Kubeflow website.Maybe move this to a README in samples folder, and hyperlink from main README

Comment (yuzisun) [reply README.md]: Moved to samples folder.

File: README.md [hunk 2]
```diff
@@ -31,7 +39,46 @@ env:
 ```
 Please refer to our [troubleshooting section](docs/DEVELOPER_GUIDE.md#troubleshooting) for recommendations and tips.
 
-### Use ###
+### KFServing in 5 Minutes
+Make sure you have [kind](https://github.com/kubernetes-sigs/kind#installation-and-usage),
```
Comment (ellis-bigelow) [inline README.md]: Why is this necessary?

Comment (ellis-bigelow) [reply README.md]: Can we at least make this optional (like a dropdown section) for those without a cluster.

File: README.md [hunk 3]
```diff
@@ -31,7 +39,46 @@ env:
...
+[kubectl](https://kubernetes.io/docs/tasks/tools/install-kubectl/#install-kubectl-on-linux),
+[kustomize](https://github.com/kubernetes-sigs/kustomize/blob/master/docs/INSTALL.md) v3.5.4+,
+[helm 3](https://helm.sh/docs/intro/install) installed before you start.(2 mins for setup)
+1) Create a quick `kind` kubernetes local cluster.(this takes 30s)
+```bash
+kind create cluster
+```
+2) Install Istio lean version, Knative Serving, KFServing all in one.(this takes 30s)
```
Comment (ellis-bigelow) [inline README.md]: + cert manager now, right?

File: docs/samples/README.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
+## KFServing Examples
+
+### KFServing InferenceService with out of the box Predictor
+[Deploy SKLearn Model with out-of-the-box InferenceService](./docs/samples/sklearn)
```
Comment (ellis-bigelow) [inline docs/samples/README.md]: IMO simplify these to just "SKLearn", "ONNX Model with ONNX Runtime", etc. Deploy and InferenceService is stated in the header.

File: README.md (hunk 3 above)
Comment (yuzisun) [reply README.md]: the quick install is not using cert manager, it is using Tommy's self signed cert gen script

**Comment by yuzisun** [comment]
@ellis-bigelow Can we merge this now?

**Comment by ellis-bigelow** [comment]
Looks like you hit a flakey test :( 
/retest
